openmdao/func_api.py

```diff
@@ -304,7 +316,7 @@ def get_input_meta(self):
         """
         if self._call_setup:
             self._setup()
-        return self._inputs.items()
+        return [it for it in self._inputs.items() if 'resid' not in it[1]]
```

Comment:
docstring still says this returns an iterator (now returns a list)

Reply:
fixed